### ISSUE TITLE
`no-console` gives a false positive when there is a `console` local

### ISSUE THREAD

**Issue by alfaproject** [palantir/tslint#3633]
### Bug Report

- __TSLint version__: `5.8.0`
- __TypeScript version__: `2.5.3`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
f(console) {
  console.log();
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-console": true
  }
}
```

#### Actual behavior
`ERROR: 2:3  no-console             Calls to 'console.log' are not allowed.`

#### Expected behavior
No error

**Comment by johnwiseheart** [comment]
This is kind of a bug I guess, but I think that it is not advisable to use a variable named `console` as it will shadow the global `console` variable. My recommendation is not to use `console` as a variable name.

**Comment by alfaproject** [comment]
That's the whole point of it tho. I don't want people to use the global `console`, and therefore that name should then be available to use freely.

**Comment by johnwiseheart** [comment]
Whether you want people to use the global console or not, a variable named ‘console’ it will still shadow the global unless you do something funky like unsetting the function. It’s a very specific use case and as such does not have a place in the main tslint ruleset, which is intended to be fairly specific and opinionated. That said, you could write your own custom rule to lint this use case.

**Comment by alfaproject** [comment]
I guess we finally got that rule https://github.com/palantir/tslint/pull/3824
